### DISCUSSION TITLE
requests to port 21114 but no docs

### DISCUSSION THREAD

**Discussion by si458** [rustdesk/rustdesk#522] [General]
after some debugging, im noticing requests trying to connect to port 21114 ?
but nowhere on github or docs mensions port 21114?
the app seems to make requests to http://myserver.com:21114/api/audit ?
im running the latest server side 1.1.5 and the latest build (built manually) from master branch

**Comment by rustdesk** [comment]
It is for future use, to remember when session is started and ended. The official name of this feature is audit log. The other vendors, Teamviewer/Anydesk etc, provide this as commercial feature. I do not know how to do in the future, but eventually we need to find a way to make both of this open source project and our life sustainable.

**Comment by LanYunDev** [comment]
@rustdesk 
Would it be possible to provide an option to support disabling this feature?
Until the feature is finalized and the documentation is updated, I'd like to not have any unnecessary unconnected connections.

**Comment by gioman** [comment]
@rustdesk not sure if I'm seeing the same, but I have deployed rustdesk (not PRO) on a server following the docs and opening the necessary ports 21115-21119. Everything works but I'm observing a flood of requests (~1 minute) on port 21114 (that is not mentioned on docs apart from something related to the pro version) from the clients that have been set up to use that server. Can you help understand what it is going on? The above answer is not clear to me. Thanks.

**Reply by fufesou** [reply]
This is a heartbeat request to get the disconnect command and latest policy from the server.

https://github.com/rustdesk/rustdesk/blob/85cafda168e927733aa36519f41de8c4bf6d8893/src/hbbs_http/sync.rs#L111


The request interval is at least `const TIME_HEARTBEAT: Duration = Duration::from_secs(15);`.

https://github.com/rustdesk/rustdesk/blob/85cafda168e927733aa36519f41de8c4bf6d8893/src/hbbs_http/sync.rs#L98

**Reply by gioman** [reply]
@fufesou thanks for the reply and the technical details. Anyway I'm still confused, what is the point of trying to get something from the server on a port that is not open (and it is clearly documented that is not needed to be open)?

**Reply by fufesou** [reply]
The Port 21114 is only for server pro.

**Reply by gioman** [reply]
@fufesou thanks, I have said it here above, I have *not*  a pro server installation.

**Reply by fufesou** [reply]
Yes. But the RustDesk client will also try request heartbeat on 21114 if it is custom client server.

**Reply by gioman** [reply]
@fufesou @rustdesk is there anything can be done to make the clients stop do that?

**Reply by Sven1403** [reply]
@fufesou @rustdesk we got the same flood of requests. Using Rustdesk self-hosted. Not Pro. Is it possible to stop this?

**Reply by Sven1403** [reply]
@gioman i spoke with a colleague from the firewall team and we tried something. We created a outbound firewall rule which blocks the remoteport 21114 combined with the rustdesk.exe. This should prevent the flood of requests. I also replaced the "RustDesk Service" firewall rule (which is created while the installation) with rules which goes on the explicit TCP/UDP ports for rustdesk instead of all ports. So far everything is working without problems on my client. We didnt deploy the rule so far... Maybe this can help you :)

**Reply by xlionjuan** [reply]
On your firewall, **drop** port 21114, not deny it.



Or your rest firewall rule should drop it, not deny it.

**Comment by Bootpanic** [comment]
It can't be hard to provide an option in the client to disable this behavior. Since quite a few people have reported this and are getting evasive answers I'm afraid there's no other option than to block the port in the client's firewall.

**Comment by VipCrazyMaster** [comment]
21114端口是给企业用户用的，如果是社区用户，在API服务填写：http://127.0.0.1:21114即可，日志噪声消失，不在有警告
